networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/Eth2Network.java

```diff
@@ -21,4 +22,6 @@
   void subscribeToAttestationCommitteeTopic(int committeeIndex);
 
   void unsubscribeFromAttestationCommitteeTopic(int committeeIndex);
+
+  void updateAttestationSubnetENRField(Bytes value);
```

Comment:
I think we've wound up mixing committeeIndex and attestation subnet in this interface with the subscribe/unsubscribe talking in terms of committee index but the attestation subnet defining subnet indices. Prior to v0.11 that was a 1-1 mapping but now that the topic index is `committeeIndex % ATTESTATION_SUBNET_COUNT` they wind up different and something has to handle cases where two committees map to the same subnet (especially when we unsubscribe from one but not the other).

I'd be tempted to define this entirely in terms of subnet index so the `AttestationTopicSubscriptions` class is responsible for the mapping from committee index to subnet index and working out which subnets we should be subscribed from that.

But I'd also be very tempted to change `updateAttestationSubnetENRField` to be more discovery system agnostic so it's just `setLongTermAttestationSubnetSubscriptions(List<Integer> subnetIndices)` and that flows all the way down to `DiscoveryNetwork` which is responsible for converting the indices to the right format to put in an ENR.  Outside of `DiscoveryNetwork` shouldn't need to know how it's represented (or really even that it goes in the ENR).  In the future it might use discv5 topic searches rather than an enr field and ideally only `DiscoveryNetwork` would have to change.

Reply:
changing `updateAttestationSubnetENRField` to `setLongTermAttestationSubnetSubscriptions(List<Integer> subnetIndices)` makes a lot of sense to me.

Reply:
However, the fact that DiscoveryNetwork is an instance of the P2PNetwork makes it impossible, as far as I understand, to keep any Discovery specific implementation detail out of P2PNetwork, DelegatingP2PNetwork, LibP2PNetwork, MockP2PNetwork etc. Any DiscoveryNetwork specific API I need to implement, I need to implement for all these classes as well.

Reply:
I think all the discovery specific stuff should be *inside* DiscoveryNetwork, not exposed by it's API.  Just like the discovery library dependency is entire inside DiscoveryNetwork.

Reply:
Makes total sense to me as discussed offline.